src/key-derivation.ts

```diff
@@ -0,0 +1,32 @@
+import BN from "bn.js";
+import {getHkdf} from "./crypto";
+// @ts-ignore
+import {SHA256} from "bcrypto";
+
+export function deriveMasterSK(ikm: Buffer): Buffer {
+    const okm = getHkdf(ikm, 48, Buffer.from("BLS-SIG-KEYGEN-SALT-", "utf-8"));
+    const okmBN = new BN(okm, 'hex', 'be');
+    const r = new BN("52435875175126190479447740508185965837690552500527637822603658699938581184513");
+    return okmBN.mod(r).toBuffer("be", 32);
+}
+
+export function deriveChildSK(parentSK: BN, index: BN): Buffer {
```

Comment:
I feel like we don't need these BNs? Shouldn't the secret key be a Buffer?

Reply:
we are doing mod operations so it must be bn. in its essence privatr key is a number

Reply:
oh I see, what you are saying, it will get updated in second PR